### ISSUE TITLE
Allow recurring ping  + interval in config ?

### ISSUE THREAD

**Issue by otaf** [gawel/panoramisk#6]
The code uses a ping method for the Manager Object, but does not check that the connection is really alive from the Manager point of view.

Maybe there is a risk that the connection dies silently, without a recurring ping from the Panoramisk App.

My 2c€ : 
* Either warn the lib user to check regularly that the connection is alive : the case in which this happens is especially when listening to events and not sending new Actions.
* or integrate a ping in the Manager code and config, with a specified interval, and forcing reconnects from the manager when the ping is KO.

**Comment by sisoftrg** [comment]
Another issue with pinger code: pinging starts in connection_made and looping forever every ten seconds.
When I close the manager, pinger don't cancelled and I have a lot of "WARNING: socket.send() raised exception." warnings.

**Comment by gawel** [comment]
You mean when you close the AMI or your panoramisk manager ? How do you close it ?

**Comment by sisoftrg** [comment]
example:
```py
ami = panoramisk.Manager()
yield from ami.connect()
try:
    res = yield from ami.send_command('sip reload')
finally:
    ami.close()
```

after this code succesfully running, I have warnings in logging output about write to closed socket every 10 seconds.

I'm fixed it by cancelling call_later() in Manager.close():

```diff
diff -ru panoramisk_/manager.py panoramisk/manager.py
--- panoramisk_/manager.py      2015-06-20 12:58:28.624925117 +0000
+++ panoramisk/manager.py       2015-06-19 08:16:14.012831258 +0000
@@ -42,6 +42,7 @@
         self.save_stream = self.config.get('save_stream')
         self.authenticated = False
         self.authenticated_future = None
+        self.pinger = None
 
     def connection_made(self, f):
         if getattr(self, 'protocol', None):
@@ -67,7 +68,7 @@
                     'Username': self.config['username'],
                     'Secret': self.config['secret']})
                 self.authenticated_future.add_done_callback(self.login)
-            self.loop.call_later(10, self.ping)
+            self.pinger = self.loop.call_later(10, self.ping)
 
     def login(self, future):
         self.authenticated_future = None
@@ -77,7 +78,7 @@
 
     def ping(self):  # pragma: no cover
         self.protocol.send({'Action': 'Ping'})
-        self.loop.call_later(10, self.ping)
+        self.pinger = self.loop.call_later(10, self.ping)
 
     def send_action(self, action, as_list=False, **kwargs):
         """Send an :class:`~panoramisk.actions.Action` to the server:
@@ -213,3 +214,6 @@
         """Close the connection"""
         if getattr(self, 'protocol', None):
             self.protocol.close()
+        if self.pinger:
+           self.pinger.cancel()
+           self.pinger = None
```

**Comment by GMLudo** [comment]
@sisoftrg your fix looks good to me.
If you have the time, you could add a test, it should be even better.

Could you open a pull request to merge your patch?

**Comment by GMLudo** [comment]
Closed via #13 PR.
Thank you again @sisoftrg
